configure

```diff
@@ -276,41 +278,39 @@ elif (args.eigen != ""):
     f.write("EIGEN = " + args.eigen + "\n")
 
 #
-# Compiler
+# Compiler and MPI library
 #
 if not args.comp in ['g++','icc','clang++']:
     raise Exception(color.red + "ERROR" + color.reset + ": Compiler must be g++, clang++, or icc")
-if (args.comp == "gcc"): f.write("COMP = GCC\n")
-elif (args.comp == "clang"): f.write("COMP = CLANG\n")
-elif (args.comp == "intel"): f.write("COMP = INTEL\n")
-message("Comp-Type", str(args.comp))
-shellmessage("Comp-Version",args.comp+" --version")
-shellmessage("Comp-MPI","mpicxx -cxx="+args.comp+" -show")
-f.write("CC = mpicxx -cxx="+args.comp+"\n")
-if args.comp == 'g++':     
-    f.write("MPI_LIB = -lgfortran -lmpich\n")
+
+if not args.alternate_mpi and not shutil.which("mpichversion"):
+    raise Exception(color.red + "ERROR" + color.reset + ": An MPICH installation was not found, to compile with another MPI library please use " + code("--alternate-mpi"))
+
+shellmessage("Compiler (" + args.comp + ")", args.comp + " --version")
+shellmessage("MPI Library","mpirun --version")
+
+env = os.environ.copy()
+env.update({'MPICH_CXX': args.comp, 'OMPI_CXX': args.comp, 'I_MPI_CXX': args.comp})
+shellmessage("MPI Options", "mpicxx -show", env)
+
+backend_comp = " MPICH_CXX=" + args.comp + " OMPI_CXX=" + args.comp + " I_MPI_CXX=" + args.comp + " "
+f.write("CC =" + backend_comp + "mpicxx\n")
```

Comment:
This changes the makefile behavior. Please ensure that the Makefile, (and pre and post includes) are unchanged in the default case.

Reply:
It _doesn't_ change the makefile behaviour. It changes what's written, not the semantics.

Only some MPI libraries support specifying the backend compiler with `-cxx`, but they all afaik, support specifying it via an environment variable. This means the same solution can be used for all libraries, simplifying the code and making it very easily extendible to other libraries, say HPE MPT. If it'd help convincing you, AMReX uses the environment variable method when needed too.

Reply:
@nmnobre Except that this breaks the metadata storage method in the Makefile. The Makefile stores several metadata macros that get passed to alamo that store information such as the compiler. I do not see how that behavior is preserved here.

Reply:
Ok, can you point me to what exactly changed?

Reply:
In the Makefile see the line that sets `METADATA_FLAGS`. This is used to pass all compilation information into alamo so it can be included with alamo output. This includes for instance the `METADATA_COMPILER` which depends on `COMP` that you have removed.